File: python/src/spaturzu/anthropic.py
"""Drop-in replacement for ``anthropic``. Swap only the import::

    - from anthropic import Anthropic
    + from spaturzu.anthropic import Anthropic
"""
from __future__ import annotations
from typing import Any, Optional

from anthropic import Anthropic as _RealAnthropic, AsyncAnthropic as _RealAsyncAnthropic
from . import get_default_spaturzu


def Anthropic(*args: Any, spaturzu: Optional[dict] = None, **kwargs: Any) -> Any:
    return get_default_spaturzu().wrap_anthropic(
        _RealAnthropic(*args, **kwargs), **(spaturzu or {})
    )


def AsyncAnthropic(*args: Any, spaturzu: Optional[dict] = None, **kwargs: Any) -> Any:
    return get_default_spaturzu().wrap_anthropic(
        _RealAsyncAnthropic(*args, **kwargs), **(spaturzu or {})
    )

File: typescript/src/providers/anthropic.ts
// Drop-in replacement for `@anthropic-ai/sdk`. Swap only the import specifier:
//   - import Anthropic from "@anthropic-ai/sdk";
//   + import Anthropic from "@spaturzu/sdk/anthropic";
import RealAnthropic from "@anthropic-ai/sdk";
import { getDefaultSpaturzu } from "../default.js";
import type { SpaturzuClientOptions } from "./openai.js";
import type { TagInput } from "../context.js";

class Anthropic extends RealAnthropic {
  /** Tag each call with an agent (supplied at runtime by the wrap proxy;
   *  declared here so it's visible to TypeScript on the drop-in client). */
  declare withAgent: (agent: string, opts?: { tags?: TagInput }) => this;

  constructor(
    opts?: ConstructorParameters<typeof RealAnthropic>[0],
    sp: SpaturzuClientOptions = {},
  ) {
    super(opts);
    return getDefaultSpaturzu().wrapAnthropic(this, sp) as Anthropic;
  }
}

export default Anthropic;
export { Anthropic };
